### PR TITLE
Add functionality to Feflow reader

### PR DESCRIPTION
**Pull request by TomFischer**
Now it is possible to read
- pure tetrehedral meshes and
- mixed element meshes (prisms, hexes and tets).

Furthermore the PR contains some very basic tests.

### CODE REVIEW

File: Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp [hunk 1]
```diff
@@ -358,8 +428,8 @@ void FEFLOWMeshInterface::readElevation(std::ifstream& in,
             // parse current line
             line_stream.str(line_string);
             line_stream >> z;
-            getline(line_stream, str_nodeList);
-            BaseLib::trim(str_nodeList, '\t');
+            std::getline(line_stream, str_nodeList);
+            boost::trim(str_nodeList);
```
Comment (norihiro-w) [inline Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp]: `trim_right` not needed here?

File: Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp [hunk 2]
```diff
@@ -376,6 +446,74 @@ void FEFLOWMeshInterface::readElevation(std::ifstream& in,
...
+            {
+                ele_nodes[k] = nodes[idx[k + n_half_nodes] - 1];
+                ele_nodes[k + n_half_nodes] = nodes[idx[k] - 1];
+            }
+            break;
+    }
+
+    switch (elem_type)
+    {
+        case MeshLib::MeshElemType::LINE:
```
Comment (norihiro-w) [inline Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp]: I thought you deal with only tetra, prism, and hex. isn't it correct?

**Comment by norihiro-w** [comment]
looks good for me-. I have only a few minor comments

File: Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp [hunk 3]
```diff
@@ -129,18 +133,44 @@ MeshLib::Mesh* FEFLOWMeshInterface::readFEFLOWFile(const std::string& filename)
...
                     readElement(fem_dim, eleType, line_string, vec_nodes));
             }
         }
+        else if (line_string.compare("VARNODE") == 0)
+        {
+            assert(!vec_nodes.empty());
+
+            vec_elements.reserve(fem_dim.n_elements);
+            if (fem_dim.n_nodes_of_element == 0) // mixed element case
+                if (!std::getline(in, line_string))
```
Comment (TomFischer) [inline Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp]: Not necessarily, it is only checked that `getline` does not fail.

File: Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp (hunk 1 above)
Comment (TomFischer) [inline Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp]: The `boost::trim` removes the `\t` like the `BaseLib::trim` above.

Comment (norihiro-w) [inline Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp]: but at other places, you use trim_right.

File: Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp (hunk 2 above)
Comment (TomFischer) [inline Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp]: I'll remove unnecessary elements.

**Comment by norihiro-w** [comment]
:+1: if tests are ok

**Pull request merged**